src/app/server/Server.cpp

```diff
@@ -125,6 +227,10 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
     VerifyOrExit(initParams.operationalKeystore != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(initParams.opCertStore != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(initParams.reportScheduler != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+#if defined CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS && defined CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION
+    VerifyOrExit(initParams.enhancedSetupFlowProvider != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(initParams.termsAndConditionsProvider != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This couples terms and conditions to ESF. ESF may be used in future with no T&Cs, and only some future ESF-based features.

For the ESF-conditional code, please ensure that null T&C provider is allowed, and simply means those are omitted.

Reply:
Terms and Conditions is a feature of ESF. ESF may contain additional features in the future. In the future, when that happens, I would imagine the change you are looking for would be made. Until then, I would prefer never-null guarantees.